### DISCUSSION TITLE
🖐️ When is image (re)pulled by dagger-engine

### DISCUSSION THREAD

**Discussion by Pymann** [dagger/dagger#12483] [Help]
### Dagger Cloud run URL

docker-container://dagger-engine

### What happened? What did you expect to happen?

I tried some examples from dagger.io.

And for me is unclear in what cases the image is repulled by the dagger-engine.
Manually pulled images seem to be ignored?

The problem is even more critical if you imagine kubernetes pod running the dagger-engine. Because there will be no info after restart of pod. Mounted docker.sock should have wished effect... but If it will repull, dagger is not of any use...

**Comment by grouville** [comment]
@Pymann, from what I understand, you've been:
1. Pulling an image in Docker using the docker CLI
2. Referencing the same image in a Dagger pipeline
3. Realizing that the pipeline would download the Dagger image even though you have it in locally in Docker

If this is what you meant, it doesn't work that way: docker's cache and dagger's cache are not related, even when you mount the socket.

> The problem is even more critical if you imagine kubernetes pod running the dagger-engine. Because there will be no info after restart of pod. Mounted docker.sock should have wished effect... but If it will repull, dagger is not of any use...

Regarding the effect on Kubernetes pod, let's ping @jlongtine and @gerhard to see how they handled that use case. It might be worth checking: https://www.youtube.com/watch?v=c93_EsedP1s

**Comment by jlongtine** [comment]
@Pymann I need to know a bit more about the architecture you're trying to use to comment. How are you trying to run the Dagger Engine in Kubernetes?
